src/FSharpVSPowerTools/Commands/PrintfSpecifiersUsageTaggerProvider.cs

```diff
@@ -89,22 +91,27 @@ public PrintfColorManager(ThemeManager themeManager, IEditorFormatMapService edi
             _editorFormatMapService = editorFormatMapService;
 
             _currentTheme = _themeManager.GetCurrentTheme();
+            _lastThemeChange = DateTime.MinValue;
         }
 
         public Color GetDefaultColor()
         {
             return _currentTheme == VisualStudioTheme.Dark ? DarkThemeColor : LightThemeColor;
         }
 
-        public void UpdateColors()
+        public void UpdateColors(bool force)
         {
             var newTheme = _themeManager.GetCurrentTheme();
 
-            if (newTheme != VisualStudioTheme.Unknown && newTheme != _currentTheme)
+            // Multiple theme change events are fired in rapid succession after the theme was changed.
+            // All of them must be processed to properly update the color scheme.
+            if (newTheme != VisualStudioTheme.Unknown &&
+                (newTheme != _currentTheme || (_lastThemeChange - DateTime.Now).TotalSeconds < 10 || force))
```

Comment:
This condition is always true because `_lastThemeChange <= DateTime.Now`, isn't this?

Or do you mean `DateTime.Now - _lastThemeChange` here?